src/hatti/ona/forms.cljs

```diff
@@ -279,32 +279,37 @@
     (cond-> field (not label) (assoc :label label))))
 
 ;; UTILITY: Form Flattening
+(defn name-label-map
+  [{:keys [children name label] :as child} prefix acc flatten-repeats?]
+  (let [full-name       (if prefix (str prefix "/" name) name)
+        langs           (when (map? label) (keys label))
+        updated-child   (with-meta
+                          (assoc child :full-name full-name) {:langs langs})
+        childless-child (dissoc updated-child :children)
+        recur'          (map #(name-label-map % full-name [] flatten-repeats?)
+                             children)]
+    (cond
+      (and
+       flatten-repeats?
+       (repeat? updated-child)) (concat (conj acc childless-child) recur')
+      (repeat? updated-child)   (conj acc
+                                      (assoc updated-child
+                                             :children
+                                             (flatten (apply concat recur'))))
+      (group? updated-child)    (concat (conj acc childless-child) recur')
```

Comment:
because this has the same right hand side as on line 294, I think this should be DRY up by having the condition be

```clojure
(or
  (group? updated-child)
  (and
    flatten-repeats?
    (repeat? updated-child)))
```